open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,736 @@
+     * @return A validation response object indicating whether the control parameters are valid.
+     */
+    public static JSONObject validateControlParameters(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        JSONObject maximumIndividualAmountResult = validateMaximumIndividualAmount(controlParameters);
+        if (!(Boolean.parseBoolean(maximumIndividualAmountResult.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return maximumIndividualAmountResult;
+        }
+
+        JSONObject maximumIndividualAmountCurrencyResult = validateMaximumIndividualAmountCurrency(controlParameters);
+        if (!(Boolean.parseBoolean(maximumIndividualAmountCurrencyResult.
+                getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return maximumIndividualAmountCurrencyResult;
+        }
+
+        JSONObject parameterDateTimeValidationResponse = validateParameterDateTime(controlParameters);
+        if (!(Boolean.parseBoolean(parameterDateTimeValidationResponse.
+                getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return parameterDateTimeValidationResponse;
+        }
+
+        // Validate Periodic Limits
+        JSONObject periodicLimitsValidationResult = validatePeriodicLimits(controlParameters);
+        if (!(Boolean.parseBoolean(periodicLimitsValidationResult.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+            return periodicLimitsValidationResult;
+        }
+
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Checks whether the given object is a valid JSONArray.
+     * This method verifies if the provided object is not null and is an instance of JSONArray.
+     * It is commonly used to validate whether a given object represents a valid JSON array.
+     *
+     * @param value The object to be checked for being a valid JSONArray.
+     */
+    public static boolean isValidJSONArray(Object value) {
+        return value instanceof JSONArray;
+    }
+
+    /**
+     * Validates the Maximum Individual Amount in the control parameters of a consent request.
+     *
+     * @param controlParameters The JSON object representing the control parameters of the consent request.
+     * @return A JSON object containing the validation response.
+     */
+    public static JSONObject validateMaximumIndividualAmount(JSONObject controlParameters) {
+
+        JSONObject validationResponse = new JSONObject();
+
+        //Validate Maximum individual amount in control parameters
+        if (controlParameters.containsKey(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT)) {
+
+            Object maximumIndividualAmount = controlParameters.
+                    get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT);
+
+            String errorMessage = String.format(ErrorConstants.INVALID_PARAMETER_MESSAGE,
```

Comment:
We dont need to create `errorMessage` variable here right? this is only needed if the validation is false, Therefor shall we move this inside `Check if the control parameter is valid` logic? (L169)

Reply:
Fixed in [e8d8048]